### PR TITLE
Prevent division by zero

### PR DESCRIPTION
**Pull request by cbastienbaron**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12 or 1.13                   |
| Bug fix?        | no                                                       |
| New feature?    | no                                                       |
| BC breaks?      | no                                                       |
| Deprecations?   | no--> |
| Related tickets |                       |
| License         | MIT                                                          |

just more robust code

### CODE REVIEW

**Comment by jakubtobiasz** [comment]
Hi @cbastienbaron 👋🏼,
Thanks for your contribution! Would you like to write a specification for this concrete case? 

Hint: `src/Sylius/Component/Core/spec/Customer/Statistics/PerChannelCustomerStatisticsSpec.php` 😅

**Comment by cbastienbaron** [comment]
hey @jakubtobiasz thanks for the hint ^^ spec has been added :crossed_fingers:

File: src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php [hunk 1]
```diff
@@ -41,6 +41,9 @@ public function getChannel(): ChannelInterface
 
     public function getAverageOrderValue(): int
     {
+        if (0 == $this->ordersCount) {
+            return 0;
+        }
```
Comment (diimpp) [inline src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php:46]: Per coding style, please add empty line before `return`

File: src/Sylius/Component/Core/spec/Customer/Statistics/PerChannelCustomerStatisticsSpec.php [hunk 2]
```diff
@@ -42,4 +42,10 @@ function it_has_an_average_value_of_an_order(): void
     {
         $this->getAverageOrderValue()->shouldReturn(2000);
     }
+
+    function it_returns_zero_average_order_value_when_order_count_is_zero(ChannelInterface $channel): void
```
Comment (diimpp) [inline src/Sylius/Component/Core/spec/Customer/Statistics/PerChannelCustomerStatisticsSpec.php]: As idea (Not asking for this change, need to check more specs to find best phrasing, but current one is good as well.)
```suggestion
    function it_has_zero_average_order_value_when_order_count_is_zero(ChannelInterface $channel): void
```

File: src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php [hunk 3]
```diff
@@ -41,6 +41,9 @@ public function getChannel(): ChannelInterface
...
         return (int) round($this->ordersValue / $this->ordersCount);
```
Comment (diimpp) [inline src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php]: Per coding style, there should be empty line before `return`.
```suggestion
        }
        
        return (int) round($this->ordersValue / $this->ordersCount);
```

**Comment by cbastienbaron** [comment]
thanks u @diimpp for review :+1:  resolved :crossed_fingers:

File: src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php [hunk 4]
```diff
@@ -41,6 +41,10 @@ public function getChannel(): ChannelInterface
 
     public function getAverageOrderValue(): int
     {
+        if (0 == $this->ordersCount) {
```
Comment (diimpp) [inline src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php]:
```suggestion
        if (0 === $this->ordersCount) {
```
Sorry, didn't notice this at first. Please use more strict comparison.

Comment (cbastienbaron) [reply src/Sylius/Component/Core/Customer/Statistics/PerChannelCustomerStatistics.php]: my bad ;) ty

**Review by diimpp — APPROVED** [review]

**Review by oallain — APPROVED** [review]

**Review by loic425 — APPROVED** [review]

**Review by Prometee — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]
Let me welcome you among Sylius contributors :)

**Comment by lchrusciel** [comment]
Thank you, @cbastienbaron!

**Comment by lchrusciel** [comment]
or it should be even merged to 1.12 :/

**Comment by cbastienbaron** [comment]
@lchrusciel Thanks you for yours kinds words 👍

**Pull request merged**